meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc

```diff
@@ -0,0 +1,96 @@
+require mender-client.inc
+
+PACKAGES =+ "mender-auth mender-update"
+
+BBCLASSEXTEND = "native"
+RDEPENDS:mender-update:class-native = ""
+DEPENDS:class-native = ""
+
+RPROVIDES:mender-update = "mender-client"
+
+# TODO - This is really all it needs to RPROVIDE for the other recipes
+RPROVIDES:mender-auth = "virtual/mender-client"
+RDEPENDS:mender-update = "mender-auth"
+
+PROVIDES += "mender-auth mender-update"
+
+DESCRIPTION = "Mender tool for doing OTA software updates."
+HOMEPAGE = "https://mender.io"
+
+SYSTEMD_SERVICE:mender-update = "mender-authd.service"
+SYSTEMD_SERVICE:mender-auth = "mender-authd.service"
+
+MENDER_CLIENT ?= "mender-updated"
+
+# Don't download the googletest source in Mender CMakefiles.txt
+EXTRA_OECMAKE:append = " -DMENDER_DOWNLOAD_GTEST=NO"
+
+inherit cmake
+
+RDEPENDS:mender-update:append:mender-growfs-data:mender-systemd = " parted util-linux-fdisk"
+RDEPENDS:mender-update:append:mender-growfs-data:mender-systemd:mender-partlabel = " util-linux-blkid"
+RDEPENDS:mender-update:append:mender-growfs-data:mender-systemd:mender-partuuid = " util-linux-blkid"
+
+
+FILES:mender-update:append:mender-image:mender-systemd = " \
+    ${systemd_unitdir}/system/mender-data-dir.service \
+    ${systemd_unitdir}/system/${MENDER_CLIENT}.service.wants/mender-data-dir.service \
+"
+
+FILES:mender-update:append:mender-growfs-data:mender-systemd = " \
+    ${bindir}/mender-resize-data-part \
+    ${systemd_unitdir}/system/mender-grow-data.service \
+    ${systemd_unitdir}/system/mender-systemd-growfs-data.service \
+    ${systemd_unitdir}/system/data.mount.wants/mender-grow-data.service \
+    ${systemd_unitdir}/system/data.mount.wants/mender-systemd-growfs-data.service \
+"
+
+FILES:mender-update:append:mender-persist-systemd-machine-id = " \
+    ${systemd_unitdir}/system/mender-systemd-machine-id.service \
+    ${systemd_unitdir}/system/${MENDER_CLIENT}.service.wants/mender-systemd-machine-id.service \
+    ${bindir}/mender-set-systemd-machine-id.sh \
+"
+
+FILES:mender-update += "\
+    ${bindir}/mender-update \
+    ${datadir}/mender/identity \
+    ${datadir}/mender/identity/mender-device-identity \
```

Comment:
These two should be part of mender-auth, not mender-update.

Reply:
Good catch - thanks!